strawberryfields/configuration.py

```diff
@@ -15,138 +15,229 @@
 This module contains the :class:`Configuration` class, which is used to
 load, store, save, and modify configuration options for Strawberry Fields.
 """
-import os
 import logging as log
+import os
 
 import toml
 from appdirs import user_config_dir
 
 log.getLogger()
 
 
-DEFAULT_CONFIG = {
-    "api": {
-        "authentication_token": "",
-        "hostname": "localhost",
-        "use_ssl": True,
-        "port": 443,
-        "debug": False}
-}
+class ConfigurationError(Exception):
+    """Exception used for configuration errors"""
+
+
+def load_config(filename="config.toml", **kwargs):
+    """Load configuration from keyword arguments, configuration file or
+    environment variables.
 
-BOOLEAN_KEYS = ("debug", "use_ssl")
+    .. note::
+
+        The configuration object (a nested dictionary) will be created based
```

Comment:
While technically true, referring to it as an object first and dictionary later can be a bit confusing, as I first think it is an object of a class
```suggestion
        The configuration dictionary will be created based
```